pkg/kube/rpc_creds_test.go

```diff
@@ -0,0 +1,60 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kube
+
+import (
+	"context"
+	"reflect"
+	"testing"
+	"time"
+
+	"istio.io/istio/galley/pkg/testing/mock"
+)
+
+func Test_tokenSupplier_GetRequestMetadata(t *testing.T) {
+	ctx := context.Background()
+	kube := mock.NewKube()
+	cli, err := kube.KubeClient()
+	if err != nil {
+		t.Fatal(err)
+	}
+	perCred, err := NewRPCCredentials(cli, "default", "default", nil, 2, 1)
+	if err != nil {
+		t.Fatal(err)
+	}
+	m1, err := perCred.GetRequestMetadata(ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	m2, err := perCred.GetRequestMetadata(ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if !reflect.DeepEqual(m1, m2) {
+		t.Fatalf("Unexpectedly getting a new tokens")
+	}
+
+	time.Sleep(time.Second)
```

Comment:
time.Sleep is not appropriate in tests, see https://github.com/istio/istio/wiki/Test-Flakes